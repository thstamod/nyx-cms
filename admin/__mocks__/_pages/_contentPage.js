export const sidebar = {
  data: {
    documentTypes: [
      {
        _id: '5e986ce27a141e6a5ca5f2ee',
        name: 'depth_0',
        __typename: 'DocumentType',
        descendants: [
          {
            documentType: {
              _id: '5e986cfb7a141e6a5ca5f2ef',
              name: 'depth_11',
              __typename: 'DocumentType',
              descendants: [],
            },
          },
          {
            documentType: {
              _id: '5e986d007a141e6a5ca5f2f0',
              name: 'depth12_with_child',
              __typename: 'DocumentType',
              descendants: [
                {
                  documentType: {
                    _id: '5e986d097a141e6a5ca5f2f1',
                    name: 'depth_2',
                    __typename: 'DocumentType',
                    descendants: [],
                  },
                },
              ],
            },
          },
        ],
      },
      {
        _id: '5e98743e707dfc01d6ce173e',
        name: 'depth0_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
      {
        _id: '5e989eee707dfc01d6ce173f',
        name: 'depth0_1_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
    ],
  },
};
export const panel = {
  data: {
    documentType: {
      _id: '5e98743e707dfc01d6ce173e',
      name: 'depth0_no child',
      parentDocumentType: null,
      compilation: [
        {
          dataType: {
            type: 'Text',
            name: 'text',
          },
          value: {
            val: 'I am a Title!!',
          },
          title: 'title',
          options: null,
        },
        {
          dataType: {
            type: 'TextArea',
            name: 'TextArea',
          },
          value: {
            val:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
          },
          title: 'big text',
          options: null,
        },
      ],
      __typename: 'DocumentType',
    },
  },
};
