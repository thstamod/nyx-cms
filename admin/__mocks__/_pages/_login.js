export default {
  data: {
    login: {
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZTcxNDczZjE4MTIxYzJmZjI0YzQwZWYiLCJlbWFpbCI6InRlc3RAbWFpbC5jb20iLCJpYXQiOjE1ODc4OTc3NTEsImV4cCI6MTU4NzkwMTM1MX0.3oIAITE622eRvD85ux1t5n1rLAhOJmyc7c7tTMXYgCA',
      tokenExpiration: 1,
    },
  },
};
