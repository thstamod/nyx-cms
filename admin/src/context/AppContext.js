import React, { createContext, useContext } from 'react';
import { userReducer, initialState } from '../redux/reducers/userReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const AppContext = createContext();

export function useAppState() {
  return useContext(AppContext);
}

export function AppStateProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(userReducer, initialState, middlewares);

  return <AppContext.Provider value={store}>{children}</AppContext.Provider>;
}
