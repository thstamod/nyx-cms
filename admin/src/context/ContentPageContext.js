import React, { createContext, useContext } from 'react';
import {
  initialState,
  contentPageReducer,
} from '../state/reducers/contentPageReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const ContentPageContext = createContext(null);

export function useContentPageState() {
  return useContext(ContentPageContext);
}

export function ContentPageProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(
    contentPageReducer,
    initialState,
    middlewares
  );

  return (
    <ContentPageContext.Provider value={store}>
      {children}
    </ContentPageContext.Provider>
  );
}
