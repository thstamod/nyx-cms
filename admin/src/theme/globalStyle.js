import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html,body {
    width:100%;
  height: 100vh;
  display: flex;
  font-family: Karla, serif;
  }
#root {
  width:100%;
  height:100%;
}

  h1,h2,h3,h4,h5 {
    font-family: Rubik, serif;
  }
`;

export default GlobalStyle;
