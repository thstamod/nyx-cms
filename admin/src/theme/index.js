const theme = {
  colors: {
    powderWhite: '#F2F3F4',
    foggyOrange: '#D6804D',
    lightPurple: '#B886A5',
    darkOrange: '#AB5C49',
    darkGrey: '#262728',
  },
  fonts: ['Karla,sans-serif', 'Rubik,sans-serif'],
  fontSizes: {
    small: '1em',
    medium: '2em',
    large: '3em',
  },
};

export default theme;
