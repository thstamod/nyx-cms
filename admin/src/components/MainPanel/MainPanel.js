import React, { useEffect } from 'react';
import withData from '../../containers/withData';
import GET_DOCTYPE_WITH_DATATYPES from '../../graphql/getDocumentTypeWithDataTypes';
import { useContentPageState } from '../../context/ContentPageContext';
import dataTypes from '../dataTypes';

const MainPanel = ({ data, loading, error, handleClick }) => {
  const [{ id }] = useContentPageState();

  useEffect(() => {
    if (id) {
      handleClick({
        variables: {
          _id: id,
        },
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [id]);

  const handleError = (err) =>
    err && (
      <div>
        Bad:
        {err.graphQLErrors.map(({ message }, i) => (
          <span key={i}>{message}</span>
        ))}
      </div>
    );

  if (data) {
    // console.log(data.documentType);
  }

  const renderDataTypes = (_data) =>
    _data.compilation &&
    _data.compilation.map((dataType) => {
      console.log(dataType);
      const { type, name } = dataType.dataType;
      const TypedComponent = dataTypes[type.toLowerCase()];
      return <TypedComponent name={name} key={Math.random()} />;
    });

  if (loading) {
    return <div>Loading...</div>;
  }
  return (
    <>
      <h1>Content page</h1>
      {data && data.documentType.name}
      {data && renderDataTypes(data.documentType)}
      {handleError(error)}
    </>
  );
};

export default withData({ query: GET_DOCTYPE_WITH_DATATYPES, lazy: true })(
  MainPanel
);
