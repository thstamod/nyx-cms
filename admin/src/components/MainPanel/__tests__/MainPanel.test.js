import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { ContentPageProvider } from '../../../context/ContentPageContext';
import theme from '../../../theme';
import MainPanel from '../MainPanel';
import { panel } from '../../../../__mocks__/_pages/_contentPage';

afterEach(cleanup);

test('MainPanel renders', () => {
  const { asFragment } = render(
    <ContentPageProvider>
      <ThemeProvider theme={theme}>
        <MainPanel data={panel.data} />
      </ThemeProvider>
    </ContentPageProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

// test('MainPanel no data', () => {
//   const { getByText } = render(
//     <ContentPageProvider>
//       <ThemeProvider theme={theme}>
//         <MainPanel data={null} />
//       </ThemeProvider>
//     </ContentPageProvider>
//   );

//   // fireEvent.click(screen.getByText('depth_0+'));
//   const el = screen.getByText('depth_0+');
//   el.click();
//   expect(getByText('depth_11')).toBeVisible();
// });

// test('MainPanel render Text box', () => {
//   const { getByText } = render(
//     <ContentPageProvider>
//       <ThemeProvider theme={theme}>
//         <MainPanel data={panel.data.documentTypes[0]} />
//       </ThemeProvider>
//     </ContentPageProvider>
//   );

//   expect(getByText('depth_11')).not.toBeVisible();
// });
