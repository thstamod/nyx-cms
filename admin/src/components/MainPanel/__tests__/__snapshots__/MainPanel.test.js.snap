// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPanel renders 1`] = `
<DocumentFragment>
  <h1>
    Content page
  </h1>
  depth0_no child
  <input
    type="text"
  />
  <input
    type="text"
  />
</DocumentFragment>
`;
